### PR TITLE
Optimize core components

### PR DESCRIPTION
**Pull request by mikedemarais**
~Prefers `StyleSheet`, memoized style objects, or inline style props over `styled-components` in order to reduce number of wrapper nodes and improve render times.~

Makes all the core shit consistently `styled-components`. also cleans up a looot

Fixes RAI-511 https://linear.app/issue/RAI-511/optimize-core-components

### CODE REVIEW

File: src/components/coin-row/SendSavingsCoinRow.js [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
+const sx = StyleSheet.create({
+  container: {
+    paddingBottom: 18,
+    paddingLeft: 15,
+    paddingTop: 6,
+  },
+  containerSelected: {
+    height: selectedHeight,
+    paddingBottom: 19,
+    paddingLeft: 15,
```
Comment (brunobar79) [inline src/components/coin-row/SendSavingsCoinRow.js:21]: Do you have any pref. of using paddingHorizontal vs paddingLeft & paddingRight when the value is the same?

File: src/components/investment-cards/UniswapInvestmentCard.js [hunk 2]
```diff
@@ -14,72 +14,77 @@ import InvestmentCardPill from './InvestmentCardPill';
...
-      ethBalance,
-      tokenBalance,
-      tokenName,
+  const headerProps = useMemo(
+    () => ({
+      color: colors.dark,
+      emoji: 'unicorn',
+      isCollapsible,
+      title: `ETH • ${tokenSymbol}`,
+      titleColor: '#E540F1',
```
Comment (brunobar79) [inline src/components/investment-cards/UniswapInvestmentCard.js:54]: I've noticed we're using this color also in `SwapDetailsState.js`, should we move it to the colors file or replace it by one of the existing ones? cc: @christianbaroni

File: src/components/Divider.js [hunk 3]
```diff
@@ -21,50 +21,51 @@ const buildInsetFromProps = inset => {
...
-  background-color: ${({ backgroundColor }) => backgroundColor || colors.white};
-  flex-shrink: 0;
-  height: ${({ horizontal, size }) => (horizontal ? size : '100%')};
-  width: ${({ horizontal, size }) => (horizontal ? '100%' : size)};
-`;
+    return (
+      <View
+        backgroundColor={backgroundColor || colors.white}
+        flexShrink={0}
+        height={horizontal ? size : '100%'}
```
Comment (osdnk) [inline src/components/Divider.js:51]: why is it faster?

**Comment by christianbaroni** [comment]
SF Pro Rounded isn't being applied everywhere

Missing icon issue @brunobar79 mentioned probably a result of this

File: src/components/Divider.js (hunk 3 above)
Comment (mikedemarais) [reply src/components/Divider.js:51]: @osdnk what specifically? 
im trying to remove `styled-components` from all of the core components that we use everywhere

lemme know if you have any suggestions / feedback!

File: src/components/layout/Row.js [hunk 4]
```diff
@@ -1,7 +1,6 @@
-import { compose, setDisplayName, withProps } from 'recompact';
+import React from 'react';
 import Flex from './Flex';
 
-export default compose(
-  setDisplayName('Row'),
-  withProps({ direction: 'row' })
-)(Flex);
+const Row = (props, ref) => <Flex direction="row" ref={ref} {...props} />;
```
Comment (osdnk) [inline src/components/layout/Row.js:4]: const Row = styled(Flex).attrs({ direction: "row" })``

export default Row

File: src/components/coin-row/SendSavingsCoinRow.js (hunk 1 above)
Comment (mikedemarais) [reply src/components/coin-row/SendSavingsCoinRow.js:21]: i think i prefer `paddingHorizontal` , nice catch on this ^

File: ios/Podfile.lock [hunk 5]
```diff
@@ -1,5 +1,5 @@
 PODS:
-  - Analytics (3.7.1)
```
Comment (osdnk) [inline ios/Podfile.lock:2]: can you revert these changes?

File: package.json [hunk 6]
```diff
@@ -199,7 +199,7 @@
     "babel-plugin-styled-components": "^1.10.7",
     "babel-plugin-transform-remove-console": "^6.9.4",
     "detox": "^15.4.2",
-    "eslint": "^6.6.0",
```
Comment (osdnk) [inline package.json:202]: and maybe changes from package.json? Seems to be not relevant to your PR

File: ios/Podfile.lock (hunk 5 above)
Comment (mikedemarais) [reply ios/Podfile.lock:2]: yup

File: package.json (hunk 6 above)
Comment (mikedemarais) [reply package.json:202]: yup ill delete the `eslint` related changes to `package.json`. i had made these changes before https://github.com/rainbow-me/rainbow/pull/600 had existed / gotten merged.

Comment (osdnk) [reply package.json:202]: I was reffing all changes in package.json. Can you move it to another PR if there're important?

Comment (mikedemarais) [reply package.json:202]: no they are relevant to this PR.
i made changes to `react-coin-icon` and `react-native-shadow-stack` which match the changes made in this PR. 

shadow stack changes = https://github.com/rainbow-me/react-native-shadow-stack/commit/50b12863d0ce2b164c414fa69280202f6b67d3e7

coin icon changes = https://github.com/mikedemarais/react-coin-icon/commit/d5f1c7f18e79234291e075058d0c99c12e0abeaf#diff-73155c59cb398e7816c5babec357fa01

**Comment by mikedemarais** [comment]
> Actually just found an issue in the deposit modal where the icons are not loading
> 
> Screen Shot 2020-04-21 at 6 04 15 PM
> 
> Also the swap details state looks weird:
> 
> Screen Shot 2020-04-21 at 6 11 06 PM

this is resolved btw

**Comment by mikedemarais** [comment]
> SF Pro Rounded isn't being applied everywhere
> 
> Missing icon issue @brunobar79 mentioned probably a result of this

this is also resolved

File: src/components/investment-cards/UniswapInvestmentCard.js [hunk 7]
```diff
@@ -14,72 +15,86 @@ import InvestmentCardPill from './InvestmentCardPill';
...
-  withHandlers({
-    onPressContainer: ({ item, onPress }) => () => {
-      if (onPress) {
-        onPress(item);
-      }
-    },
-  })
-);
+  const handleOpenExpandedState = useCallback(() => {
+    navigate('ExpandedAssetScreen', {
```
Comment (brunobar79) [inline src/components/investment-cards/UniswapInvestmentCard.js:42]: Routes.

File: src/components/exchange/ExchangeModalHeader.js [hunk 8]
```diff
@@ -71,15 +67,13 @@ const ExchangeModalHeader = ({ onPressDetails, showDetailsButton, title }) => {
...
+      </Text>
       <InfoButtonTransition ref={ref} transition={transition}>
         {showDetailsButton && (
-          <InfoButton onPress={onPressDetails} useNativeDriver>
-            <Icon
-              {...position.sizeAsObject(InfoIconSize)}
-              color={colors.alpha(colors.blueGreyDark, 0.3)}
-              name="info"
-            />
+          <InfoButton as={TouchableScale} onPress={onPressDetails}>
```
[the remark this answers is not in the document]
Comment (mikedemarais) [reply src/components/exchange/ExchangeModalHeader.js:75]: @brunobar79 https://styled-components.com/docs/api#as-polymorphic-prop

File: package.json [hunk 9]
```diff
@@ -94,6 +94,7 @@
     "lottie-react-native": "^3.3.2",
     "match-sorter": "^4.0.2",
     "nanoid": "^2.1.6",
+    "node-emoji": "omnidan/node-emoji#v2",
```
[the remark this answers is not in the document]
Comment (mikedemarais) [reply package.json:97]: @brunobar79 i dont think it will! 

basically we were using a package previously called `react-native-emoji` and it was using an old version of this library

i decided that using `react-native-emoji` was not worth it, and we now implement the functionality ourselves by using this `node-emoji` lib inside our `text/Emoji.js` component
